3-scalability/README.md

```diff
@@ -0,0 +1,427 @@
+
+**Best for:**
+- Large repository counts (> 10,000 repos)
+- Enterprise production environments
+- When you need automatic scaling and parallel processing
+- Fully managed infrastructure with minimal operational overhead
+
+## Overview
+
+This example deploys mass-ingest at scale using:
+- **AWS Batch** - Managed batch processing with automatic scaling
+- **ECS** - Container orchestration
+- **EventBridge Scheduler** - Automated daily runs
+- **Parallel workers** - Process repository partitions concurrently
+
+Architecture:
+1. **Chunk job** - Divides repos.csv into partitions and submits processor jobs
+2. **Processor jobs** - Multiple workers process different repository ranges in parallel
+3. **Scheduler** - Triggers the chunk job daily (configurable)
+
+## Prerequisites
+
+- AWS account with appropriate permissions
+- Terraform installed (>= 1.0)
+- Docker for building the image
+- AWS CLI configured
+- ECR repository or Docker registry
+- repos.csv file with repositories to ingest
+
+## Quick start
+
+### 1. Build and push Docker image
+
+```bash
+# Build the image from repository root
+docker build -t mass-ingest:latest ..
+
+# Tag for your registry
+docker tag mass-ingest:latest <your-registry>/mass-ingest:latest
+
+# Push to registry
+docker push <your-registry>/mass-ingest:latest
+```
+
+For AWS ECR:
+```bash
+aws ecr get-login-password --region us-east-1 | docker login --username AWS --password-stdin <account-id>.dkr.ecr.us-east-1.amazonaws.com
+
+docker build -t mass-ingest:latest ..
+docker tag mass-ingest:latest <account-id>.dkr.ecr.us-east-1.amazonaws.com/mass-ingest:latest
+docker push <account-id>.dkr.ecr.us-east-1.amazonaws.com/mass-ingest:latest
+```
+
+### 2. Configure Terraform variables
+
+Create `terraform/terraform.tfvars`:
+
+```hcl
+name                      = "mass-ingest"
+vpc_id                    = "vpc-xxxxx"
+subnet_ids                = ["subnet-xxxxx", "subnet-yyyyy"]
+image_registry            = "<your-registry>/mass-ingest"
+image_tag                 = "latest"
+moderne_tenant            = "app"  # or your tenant name
+moderne_token             = "arn:aws:secretsmanager:region:account:secret:moderne-token"
+moderne_publish_url       = "https://artifactory.example.com/artifactory/moderne-ingest/"
+moderne_publish_user      = "arn:aws:secretsmanager:region:account:secret:publish-user"
+moderne_publish_password  = "arn:aws:secretsmanager:region:account:secret:publish-password"
+
+default_tags = {
+  Environment = "production"
+  Project     = "mass-ingest"
+}
+```
+
+### 3. Store secrets in AWS Secrets Manager
+
+```bash
+aws secretsmanager create-secret \
+  --name mass-ingest/moderne-token \
+  --secret-string "your-moderne-token"
+
+aws secretsmanager create-secret \
+  --name mass-ingest/publish-user \
+  --secret-string "your-artifactory-user"
+
+aws secretsmanager create-secret \
+  --name mass-ingest/publish-password \
+  --secret-string "your-artifactory-password"
+```
```

Comment:
We probably want to move this before the terraform step, so that they'll have the secret ARN when they set up the tfvars.